src/sentry/conduit/tasks.py

```diff
@@ -0,0 +1,165 @@
+
+
+@instrumented_task(
+    name="sentry.conduit.tasks.stream_demo_data",
+    namespace=conduit_tasks,
+    processing_deadline_duration=TASK_PROCESSING_DEADLINE_SECONDS,
+    silo_mode=SiloMode.REGION,
+)
+def stream_demo_data(org_id: int, channel_id: str) -> None:
+    """Asynchronously stream data to Conduit."""
+    token = generate_jwt(subject="demo")
+    logger.info(
+        "conduit.stream_demo_data.started", extra={"org_id": org_id, "channel_id": channel_id}
+    )
+    sequence = 0
+    start_publish_request = PublishRequest(
+        channel_id=channel_id,
+        message_id=str(uuid4()),
+        sequence=sequence,
+        client_timestamp=get_timestamp(),
+        phase=Phase.PHASE_START,
+    )
+    publish_data(org_id, start_publish_request, token)
+    sequence += 1
+
+    for i in range(NUM_DELTAS):
+        payload = Struct()
+        payload.update({"value": str(i)})
+        delta_publish_request = PublishRequest(
+            channel_id=channel_id,
+            message_id=str(uuid4()),
+            sequence=sequence,
+            client_timestamp=get_timestamp(),
+            phase=Phase.PHASE_DELTA,
+            payload=payload,
+        )
+        publish_data(org_id, delta_publish_request, token)
+        sequence += 1
+        time.sleep(SEND_INTERVAL_SECONDS)
+
+    end_publish_request = PublishRequest(
+        channel_id=channel_id,
+        message_id=str(uuid4()),
+        sequence=sequence,
+        client_timestamp=get_timestamp(),
+        phase=Phase.PHASE_END,
+    )
+    publish_data(org_id, end_publish_request, token)
+    logger.info(
+        "conduit.stream_demo_data.ended", extra={"org_id": org_id, "channel_id": channel_id}
+    )
+
+
+def generate_jwt(
+    subject: str, issuer: str | None = None, audience: str | None = None, secret: str | None = None
+) -> str:
+    """
+    Generate a JWT token for the Conduit publish API.
+
+    Uses HS256 algorithm with a 5 minute expiration.
+    """
+    if issuer is None:
+        issuer = settings.CONDUIT_PUBLISH_JWT_ISSUER
+    if audience is None:
+        audience = settings.CONDUIT_PUBLISH_JWT_AUDIENCE
+    if secret is None:
+        secret = settings.CONDUIT_PUBLISH_SECRET
+        if secret is None:
+            raise ValueError("CONDUIT_PUBLISH_SECRET not configured")
+    claims = {
+        "sub": subject,
+        "iss": issuer,
+        "aud": audience,
+        "exp": int(time.time()) + JWT_EXPIRATION_SECONDS,
+    }
+    return jwt.encode(claims, secret, algorithm="HS256")
+
+
+def should_retry_publish(attempt: int, exception: Exception) -> bool:
+    return attempt < PUBLISH_REQUEST_MAX_RETRIES and isinstance(exception, RequestException)
+
+
+publish_retry_policy = ConditionalRetryPolicy(
+    should_retry_publish,
+    exponential_delay(0.5),
+)
+
+
+def publish_data(
+    org_id: int,
+    publish_request: PublishRequest,
+    token: str,
+    publish_url: str | None = None,
+) -> Response:
```

Comment:
Would something like this be part of the conduit client libraries? I could see other client applications needing message publishing with retries.